File: LinkedList/ReverseLinkedList.java
/*
Reverse a singly linked list.
*/

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ListNode reverseList(ListNode head) {
        if(head == null) return null;
        ListNode p = head.next;
        head.next = null;
        while(p != null){
            ListNode next = p.next;
            p.next = head;
            head = p;
            p = next;
        }
        return head;
    }
}

File: LinkedList/ReverseLinkedList.js
/*
Reverse a singly linked list.
*/

/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */
/**
 * @param {ListNode} head
 * @return {ListNode}
 */
var reverseList = function(head) {
    if(head === null) return head;
    var p = head.next;
    head.next = null;
    while(p !== null){
        var next = p.next;
        p.next = head;
        head = p;
        p = next;
    }
    return head;
};